IPLD-Data-Model-v1.md

```diff
@@ -1,17 +1,108 @@
-# [WIP] IPLD Data Model
+IPLD Data Model
+===============
 
-## Simple Types
+The IPLD Data Model is a core part of the IPLD specification,
+which describes what data is representable in IPLD --
+for example, booleans, integers, textual strings, maps and lists, etc.
+
+While the Data Model describes these representations in the abstract,
+[Codecs](Codecs) specify exactly how these data are transcribed into serialized bytes.
+(Another component of the IPLD specifications, [Schemas](schema), provide
+additional optional tooling on top of the Data Model which can further refine,
+describe, and constrain the range of acceptable data values.)
+
+
+Kinds
+-----
+
+The following is the list of essential _kinds_ (or more formally, _representation kinds_)
+of data representable in the IPLD Data Model:
 
-* Boolean
 * Null
+* Boolean
+* Integer
+* Float
 * String
+* Bytes
+* List
+* Map
+* Link
+
+(Note that we use the term "_kinds_" here to disambiguate this from "_types_",
+which is a term we'll use at the [Schemas](schema) level.)
+
+The _recursive kinds_ are:
+
+* List
+* Map
+
+The _scalar kinds_ (the complement of recursive) are:
+
+* Null
+* Boolean
 * Integer
 * Float
-* Array
-* Object (Hash Map)
-* Binary
+* String
+* Bytes
+* Link
+
+(Note that [Schemas](schema) introduce a few more kinds -- when clarification is necessary,
+these Data Model kinds can be called the "_representation kinds_",
+while the additional kinds introduced in the Schema layer are "_perceived kinds_".)
+
+### Kinds Reference
+
+Each of the following sections provides details about each of the kinds
+introduced in the summary table above.
+
+#### Null kind
+
+Null is a scalar kind.  Its cardinality is one -- the only value is 'null'.
 
-## Link Type
+#### Boolean kind
 
-This type represents a link to another IPLD Block. The link reference
+Boolean is a scalar kind.  Its cardinality is two -- either the value 'true' or the value 'false'.
+
+#### Integer kind
```

Comment:
Should we be providing bounds (or safe bounds at least) for integers and floats? I can imagine extensions to the schema that introduce "big" number types that are different ways of encoding with the base kinds, strings, byte arrays, tuples of integers, etc.

Reply:
Probably but that's a super huge issue, both of nuclear-reaction-design complexity and strangely popularized enough that it still can attract bikeshed-painting levels of casual interjections -- I'm utterly terrified of getting hung up on that and don't wanna talk about it in this PR :)

Reply:
I’m fine leaving it ambiguous for now and maybe going back and scaling it *down* to least common denominator once we have a known way in Schema land to do improved number types.